pubsublite/internal/wire/periodic_task.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package wire
+
+import (
+	"time"
+)
+
+// periodicTask manages a recurring background task.
+type periodicTask struct {
+	period  time.Duration
+	ticker  *time.Ticker
+	stop    chan struct{}
+	stopped bool
+	task    func()
+}
+
+func newPeriodicTask(period time.Duration, task func()) *periodicTask {
+	return &periodicTask{
+		ticker: time.NewTicker(period),
+		stop:   make(chan struct{}),
+		period: period,
+		task:   task,
+	}
+}
+
+// Start the polling goroutine.
+func (pt *periodicTask) Start() {
+	go pt.poll()
```

Comment:
Does it matter that `Start` can be called multiple times?

Reply:
Yeah, I considered handling this, but didn't think it was strictly necessary. The callers of this need to prevent multiple Starts anyway (along with other startup operations).

Added a comment in case future usage doesn't guard against duplicate starts.

Reply:
Obsolete